RELEASE-NOTES.md

```diff
@@ -63,6 +64,7 @@ to use Open Cascade's file I/O capabilities in support of Quest applications.
 
 ###  Fixed
 - Fixes compilation issue with RAJA@2024.07 on 32-bit Windows configurations. 
+- Minor bugfix to `primal::intersect(segment, ray)` to better handle cases when segment and ray overlap.
```

Comment:
Looks like this needs to be moved up a line or down a few since it's in the middle of the previous item